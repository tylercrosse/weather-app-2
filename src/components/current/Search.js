import React from "react";
import ReactDOM from "react-dom";
import PropTypes from "prop-types";
import RecentSearches from "./RecentSearches";
import SearchBar from "../SearchBar.js";

class Search extends React.Component {
  componentDidMount() {
    document.addEventListener(
      "click",
      this.handleClickOutside.bind(this),
      { capture: true, once: true }
    );
  }

  componentWillUnmount() {
    document.removeEventListener(
      "click",
      this.handleClickOutside.bind(this),
      true
    );
  }

  handleClickOutside(event) {
    const domNode = ReactDOM.findDOMNode(this.refs.search);

    if (!domNode || !domNode.contains(event.target)) {
      this.props.hideSearch();
    }
  }
  render() {
    return (
      <div ref="search" className="currentWeather__location">
        <SearchBar geocode={this.props.geocode} />
        <RecentSearches
          locations={this.props.locations}
          fetchForecast={this.props.fetchForecast}
        />
      </div>
    );
  }
}

Search.propTypes = {
  locations: PropTypes.object.isRequired,
  geocode: PropTypes.func.isRequired,
  fetchForecast: PropTypes.func.isRequired,
  hideSearch: PropTypes.func.isRequired
};

export default Search;
