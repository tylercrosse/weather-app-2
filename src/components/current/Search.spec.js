import React from 'react';
import { shallow } from 'enzyme';
import toJson from "enzyme-to-json";
import Search from './Search';

const setup = () => {
  const props = {
    locations: {
      ChIJVTPokywQkFQRmtVEaUZlJRA: {
        time: 1499098513,
        id: 'ChIJVTPokywQkFQRmtVEaUZlJRA',
        address: 'Seattle, WA, USA',
        lat: 47.6062095,
        lng: -122.3320708
      },
      'ChIJW-T2Wt7Gt4kRKl2I1CJFUsI': {
        time: 1499061460,
        id: 'ChIJW-T2Wt7Gt4kRKl2I1CJFUsI',
        address: 'Washington, DC, USA',
        lat: 38.9071923,
        lng: -77.03687070000001
      },
    },
    fetchForecast: jest.fn(),
    geocode: jest.fn(),
    hideSearch: jest.fn()
  }

  const component = <Search {...props} />;
  const wrapper = shallow(component);

  return {
    props,
    wrapper
  }
}

describe('<Search />', () => {
  it('should render correctly', () => {
    const { wrapper } = setup();
    expect(toJson(wrapper)).toMatchSnapshot();
  });
})
