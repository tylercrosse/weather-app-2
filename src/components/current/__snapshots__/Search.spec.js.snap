// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> should render correctly 1`] = `
<div
  className="currentWeather__location"
>
  <SearchBar
    geocode={[Function]}
  />
  <RecentSearches
    fetchForecast={[Function]}
    locations={
      Object {
        "ChIJVTPokywQkFQRmtVEaUZlJRA": Object {
          "address": "Seattle, WA, USA",
          "id": "ChIJVTPokywQkFQRmtVEaUZlJRA",
          "lat": 47.6062095,
          "lng": -122.3320708,
          "time": 1499098513,
        },
        "ChIJW-T2Wt7Gt4kRKl2I1CJFUsI": Object {
          "address": "Washington, DC, USA",
          "id": "ChIJW-T2Wt7Gt4kRKl2I1CJFUsI",
          "lat": 38.9071923,
          "lng": -77.03687070000001,
          "time": 1499061460,
        },
      }
    }
  />
</div>
`;
